### ISSUE TITLE
Lock is global - change to local (per nextcloud instance) possible

### ISSUE THREAD

**Issue by james-cook** [matiasdelellis/facerecognition#151]
This is really a low prio question - I have multiple nextcloud instances on the server. 
Running **face:background_job** for e.g. 2 instances at the same time is not possible because the 2nd call can not acquire the global lock:
```
cd <nextcloud instance>
sudo -u www-data php occ face:background_job
1/10 - Executing task CheckRequirementsTask (Check all requirements)
2/10 - Executing task CheckCronTask (Check that service is started from either cron or from command)
3/10 - Executing task LockTask (Acquire lock so that only one background task can run)
        Background job is already running. Quitting
Task LockTask signalled we should not continue, bailing out
```

This is probably about resources overall - but is there a technical reason **face:background**_job cannot run on separate nextcloud instances on the same machine.
I'd be interested to fine out!

**Comment by stalker314314** [comment]
Yes, unfortunately, lock is global (per machine). Here is implementation:
https://github.com/matiasdelellis/facerecognition/blob/master/lib/BackgroundJob/Tasks/LockTask.php
and I didn't find good rationale in [functional spec](https://github.com/matiasdelellis/facerecognition/wiki/Functional-spec#background) why do we take it - it was more to avoid potential concurrency issues and not to waste machine resources, than anything other. While it would be trivial to take lock per instance (we can just include instance name in the name of the lock), I am more interested if you would be fine with that. Thing is, face recognition can take massive amount of resources on your machine and not sure if you want to run them all at the same time in parallel?

If you want them sequentially on same machine, you can always achieve that by crafting proper cron jobs (start face recognition for one instance, then for other...). Otherwise, we can change logic of lock not to fail immediately, but to do busy wait (in loop, until lock could be taken, up until to timeout).

If you really, really want them in parallel, we can add additional parameter (let's say `--allow-multiple-jobs-per-server` or something like that). That, I think, should never be default, but you (and people with same needs) at least can use it
